python/src/aiconfig/editor/client/src/components/aiconfigReducer.ts

```diff
@@ -37,6 +38,12 @@ export type UpdatePromptNameAction = {
   name: string;
 };
 
+export type UpdatePromptModelAction = {
```

Comment:
When we dispatch the "UPDATE_PROMPT_MODEL" action, that action is implicitly typed as this `UpdatePromptModelAction` type. We add it to `MutateAIConfigAction` above and that does two things:
1. Makes it part of the `AIConfigReducerAction` via the union for that type, which means our reducer below will require a `case "UPDATE_PROMPT_MODEL"` to handle it to ensure the state is update -- this is the automatic mapping you're referring to
2. Allows us to dispatch `CONSOLIDATE_AICONFIG` after we get the response, in order to update the client state with response state (not done here because we don't have an aiconfig response to do that with just yet)

Reply:
Nice, thanks for the explanation!